### ISSUE TITLE
[Feature]: Support Count API

### ISSUE THREAD

**Issue by jiaoew1991** [milvus-io/milvus-sdk-rust#64]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

Milvus support `Count` API from 2.3.0, we need to support it,
more details about the it can be found here https://milvus.io/docs/query.md#Count-entities

**Comment by Hardik-7892** [comment]
Hi, I'd like to pick this up — I've already implemented and tested it locally, so I'd appreciate being assigned to avoid duplicate work.

Summary of what's done:
- Client::count() (whole-collection) and Client::count_with_expr() (filtered, honoring QueryOptions like partition names/consistency) in src/v1/query.rs, using the existing count(*) Query aggregate — no proto bump needed since the pinned proto has no dedicated Count RPC.
- Unknown/empty results degrade to 0 instead of panicking.
- Integration test count_api added in tests/v1/milvus26.rs, passing against a live Milvus v2.6.19.
- Commit is DCO-signed.

I'll open the PR from my fork shortly and drop the link here. Thanks!

**Comment by xiaofan-luan** [comment]
I think we don't need a count api?

it should be already part of aggregation.

**Comment by Hardik-7892** [comment]
Agreed, count(\*) is already reachable via the Query aggregation, and that's exactly what PR #133 does under the hood: count()/count_with_expr() are thin wrappers around the existing aggregate, matching pymilvus.num_entities and milvus-sdk-go Count. No new RPC or proto change. If the team prefers, I'm happy to drop the public API and keep just the documented count(\*) Query pattern instead.

**Comment by yhmo** [comment]
ClientV2 already supports it: https://github.com/milvus-io/milvus-sdk-rust/pull/129
